server/util/lru/lru.go

```diff
@@ -134,10 +134,9 @@ func (c *LRU) PushBack(key, value interface{}) bool {
 	if !ok {
 		return false
 	}
-	// Check for existing item
+	// Remove any existing item.
```

Comment:
Yeah, it looks like disk_cache_test is failing. Presumably, the test doesn't like that the OnEvict function is deleting the existing file. Will think about this a bit more but it might make sense to add an `EqualsFn` and skip removing the existing item if calling `EqualsFn(newValue, existingValue)` on the same key returns true (and just move it to the front of the LRU)